Source/URedis/Private/URedis.cpp

```diff
@@ -61,4 +61,26 @@ void FURedis::Rename(FStringView key, FStringView newKey) const {
                       TCHAR_TO_UTF8(newKey.GetData()));
 }
 
+#pragma region Pub/Sub
+
+uint64 FURedis::Publish(FStringView channel, FStringView message) const {
+    return _instance->publish(TCHAR_TO_UTF8(channel.GetData()),
+                              TCHAR_TO_UTF8(message.GetData()));
+}
+
+void FURedis::Subscribe(FStringView channel,
+                        void (*MsgCallback)(FStringView channel,
+                                            FStringView message)) {
+    _instance->subscriber().subscribe(TCHAR_TO_UTF8(channel.GetData()));
+    _instance->subscriber().on_message(MsgCallback);
```

Comment:
@galeone 

Hi,
How are you?
Here I've got a question about feature realization. Function `on_message` of **redis-plus-plus** library requires a callback function type of:
```cpp
std::function<void (std::string channel, std::string msg)>
```
And in Unreal Engine code we are using `FStringView` instead of regular string. I've created callback function type of:
```cpp
void (*MsgCallback)(FStringView channel, FStringView message)
```
which is not compatible to the original callback function.
Do you have any ideas or possible solutions for this problem?

Reply:
Hi @EliLentz sorry for the late reply!

A couple of suggestions: when passing callback functions it's better to avoid using string views and instead use the standard strings (in our case FString - and passing then by copy). The reason is that we don't have idea about the exact moment in which the callback will be invoked, and thus using a view-type there's the risk that the passed data has been destroyed.

Another suggestion is to completely avoid C-style functions. We have std::function and TFunction for unreal.

So, the ideal signature for the callback must be something like:

````cpp
TFunction<void(FStiring, FString)>
````

However, functions with different signatures are not compatible (you can't convert them) - so I guess the easiest solution is to add a wrapper that converts the function call.


```cpp
void FURedis::Subscribe(FStringView channel, TFunction<void(FStiring, FString)> callback) {
    _instance->subscriber().subscribe(TCHAR_TO_UTF8(channel.GetData()));
    
    auto wrapper = [callback](std::string channel, std::string message) -> void {
         callback(TCHAR_TO_UTF8(*channel), TCHAR_TO_UTF8(*message);
    };
    _instance->subscriber().on_message(wrapper);
```

I haven't tested it - but I guess it should work. Let me know if it solves the issue